runtime/src/main/scala/akka/grpc/netty/NettyCircuitBreakerInterceptor.scala

```diff
@@ -0,0 +1,57 @@
+package akka.grpc.netty
+
+import akka.pattern.CircuitBreaker
+import io.grpc.CallOptions
+import io.grpc.Channel
+import io.grpc.ClientCall
+import io.grpc.ClientInterceptor
+import io.grpc.ClientInterceptors
+import io.grpc.ForwardingClientCallListener
+import io.grpc.Metadata
+import io.grpc.MethodDescriptor
+import io.grpc.Status
+
+/**
+ * A `io.grpc.ClientInterceptor` that prevents downstream overload wrapping the calls with a circuit breaker.
+ */
+class NettyCircuitBreakerInterceptor(breaker: CircuitBreaker) extends ClientInterceptor {
+
+  private class Listener[RespT](delegate: ClientCall.Listener[RespT])
+      extends ForwardingClientCallListener.SimpleForwardingClientCallListener[RespT](delegate) {
+    override def onClose(status: Status, trailers: Metadata): Unit = {
+      if (status.isOk) {
+        breaker.succeed()
```

Comment:
One thing we'll miss when using breaker.succeed/fail is the callTimeout feature. Are there timeout's at the gRCP level for req/response calls?

If we want to include callTimeout we could use `withCircuitBreaker` and create a Promise when call is started and complete/fail that here. That's more overhead than breaker.succeed/fail so only if we think callTimeout is important.